executor/adapter.go

```diff
@@ -204,7 +204,40 @@ type ExecStmt struct {
 	isSelectForUpdate bool
 	retryCount        uint
 
-	outputNames []*types.FieldName
+	// OutputNames will be set if using cached plan
+	OutputNames []*types.FieldName
+}
+
+// GetPointRecord short path for point exec directly from plan, keep only necessary steps
+func (a *ExecStmt) GetPointRecord(ctx context.Context, is infoschema.InfoSchema,
+	sctx sessionctx.Context) (*recordSet, error) {
+	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
+		span1 := span.Tracer().StartSpan("ExecStmt.GetPointRecord", opentracing.ChildOf(span.Context()))
+		span1.LogKV("sql", a.OriginText())
+		defer span1.Finish()
+		ctx = opentracing.ContextWithSpan(ctx, span1)
+	}
+	var err error
+	var startTs uint64 = math.MaxUint64
+	err = sctx.InitTxnWithStartTS(startTs)
+	if err != nil {
+		return nil, err
+	}
+	sctx.GetSessionVars().StmtCtx.Priority = kv.PriorityHigh
+	b := newExecutorBuilder(sctx, is)
+	exec := b.build(a.Plan)
```

Comment:
How about calling `a.buildExecutor()` to build the operator tree from the execution plan?

Reply:
here just keep necessary steps to make it clear for the short path